src/graphql/schema/collection/createCollectionQueryFieldEntries.ts

```diff
@@ -93,11 +93,11 @@ function createCollectionPrimaryKeyField <TValue, TKey>(
       },
     },
 
-    async resolve (_source, args, context): Promise<mixed> {
+    async resolve (_source, args, context, resolveInfo): Promise<mixed> {
       const result = idSerde.deserialize(inventory, args[options.nodeIdFieldName] as string)
       if (result.collection !== collection) throw new Error(`The provided id is for collection '${result.collection.name}', not the expected collection '${collection.name}'.`)
       if (!keyType.isTypeOf(result.keyValue)) throw new Error(`The provided id is not of the correct type.`)
-      const value = await collectionKey.read!(context, result.keyValue)
+      const value = await collectionKey.read!(context, result.keyValue, resolveInfo)
```

Comment:
This here is also where the nice GraphQL/Postgres separation breaks down. If possible, it would be great to pass in a `ReadOptimizations` object like I mention above.